server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -1537,6 +1562,25 @@ ClusterStateApplyResponse getClusterStateApplyResponse() {
                 return clusterStateApplyResponse;
             }
 
+            Runnable onNode(Runnable runnable) {
+                final Runnable wrapped = onNodeLog(localNode, runnable);
+                return new Runnable() {
+                    @Override
+                    public void run() {
+                        if (clusterNodes.contains(ClusterNode.this) == false) {
```

Comment:
Isn't this change kind of removing a class of problems where a connection is lost while a transport action is running from our coverage?

Reply:
no, we have (prior to this PR) never removed a `ClusterNode`. What this achieves is that a dead node (i.e. a node that is shut down) stops running any tasks. One option would be to actively remove the node's tasks from the DeterministicTaskQueue (which would requires us to be able to identify the tasks), or what we do here, to just turn those tasks into NOOPs.